library/VTKExtensions/Rendering/vtkF3DRenderer.cxx

```diff
@@ -187,15 +188,18 @@ void vtkF3DRenderer::Initialize(const std::string& up)
   this->GridInfo = "";
 
   // Importer rely on the Environment being set, so this is needed in the initialization
-  if (up.size() == 2)
+  const std::regex re("([-+]?)([XYZ])", std::regex_constants::icase);
+  std::smatch match;
+  if (std::regex_match(up, match, re))
   {
-    char sign = up[0];
-    char axis = std::toupper(up[1]);
-    if ((sign == '-' || sign == '+') && (axis >= 'X' && axis <= 'Z'))
+    const float sign = match[1].str() == "-" ? -1.0 : +1.0;
+    const int index = std::toupper(match[2].str()[0]) - 'X';
+    if (index >= 0 && index < 3)
```

Comment:
Can this condition fail? If not, it should be an assert.

Reply:
Good point. Straight up `assert(index >= 0 && index < 3)` or is there a more c++ way?

Reply:
looks good to me